wagtail/images/fields.py

```diff
@@ -124,10 +124,11 @@ def to_python(self, data):
         if f is None:
             return None
 
-        # We need to get a file object for Pillow. We might have a path or we might
-        # have to read the data into memory.
+        # We need to get a file object for Pillow. When we get a path, we need to open
+        # the file first. And we have to read the data into memory to pass to Willow.
         if hasattr(data, "temporary_file_path"):
-            file = data.temporary_file_path()
+            with open(data.temporary_file_path(), "rb") as fh:
+                file = BytesIO(fh.read())
```

Comment:
Question: If `willow.Image.open` expected a file-like object, why not just pass it `fh` here? This way there's no need to necessarily load the entire file into memory, and it allows Willow to handle its own buffering / seeking.

Reply:
I didn't like the alternative 😂

```python
        if hasattr(data, "temporary_file_path"):
            file = open(data.temporary_file_path(), "rb")
            has_file_handle = True
        else:
            has_file_handle = False
            if hasattr(data, "read"):
                file = BytesIO(data.read())
            else:
                file = BytesIO(data["content"])

        try:
            # Annotate the python representation of the FileField with the image
            # property so subclasses can reuse it for their own validation
            f.image = willow.Image.open(file)
            f.content_type = image_format_name_to_content_type(f.image.format_name)

        except Exception as exc:
            # Willow doesn't recognize it as an image.
            raise ValidationError(
                self.error_messages["invalid_image"],
                code="invalid_image",
            ) from exc

        if has_file_handle:
            file.close()
```

Reply:
ok, changed in 976554a

Reply:
ok, reverted that change as we can't close the fh until late in the process, and in places we raise various validation errors so it becomes quite messy